proposals/threads/ConditionalSegmentInitialization.md

```diff
@@ -74,17 +74,29 @@ requires two modules where one should be enough.
 
 ## Proposal: New instructions to initialize data and element segments
 
-Similar to solution 2, we repurpose the memory index as a flags field. Unlike
-solution 2, the flags field specifies whether this segment is _inactive_. An
-inactive segment will not be automatically copied into the memory or table on
-instantiation, and must instead be applied manually using two new instructions:
-`init_memory` and `init_table`.
+The [binary format for the data section](https://webassembly.github.io/spec/binary/modules.html#data-section)
+currently has a collection of segments, each of which has a memory index, an
+initializer expression for its offset, and its raw data.
 
-When the least-significant bit of the flags field is `1`, the segment is
-inactive. The rest of the bits of the flags field must be zero.
+Since WebAssembly currently does not allow for multiple memories, the memory
+index must be zero. We can repurpose this field as a flags field.
+
+When the least-significant bit of the flags field is `1`, this segment is
+_inactive_. An inactive segment will not be automatically copied into the
+memory or table on instantiation, and must instead be applied manually using
+the following new instructions:
+
+* `mem.init`: copy a region from a data segment
+* `table.init`: copy an region from an element segment
 
 An inactive segment has no initializer expression, since it will be specified
-as an operand to `init_memory` or `init_table`.
+as an operand to `mem.init` or `table.init`.
+
+Segments (either active or inactive) can also be discarded by using the
+following new instructions:
```

Comment:
Gosh, I'm not sure if it's a good idea to be able to drop active memories.  Suppose you can, and do.  Then you instantiate the module again.  What happens then?  A trap?  (That would be the interpretation of the prose further down.)

Reply:
Yeah, I wasn't sure about this one. I was thinking it would be a nice optimization to be able to discard to active segments (say, if you know that there will never be another instance), but perhaps we should be more conservative.

Reply:
I'd err on the side of conservative.

Reply:
OK, changed.